### PR TITLE
Storybook: Create Welcome page

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
- Create a Welcome page
- Add Blog page
- Migrate all pages from websites GUIDE section 
Screenshot 2023-03-02 at 21 59 14

## Related Issue
https://github.com/recharts/recharts/issues/3406

## Screenshots (if appropriate):
Screenshot 2023-03-02 at 22 09 15

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/Getting started.mdx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+<LineChart width={600} height={300} data={data} margin={{ top: 5, right: 20, bottom: 5, left: 0 }}>
+  <Line type="monotone" dataKey="uv" stroke="#8884d8" />
+  <CartesianGrid stroke="#ccc" strokeDasharray="5 5" />
+  <XAxis dataKey="name" tick={<CustomizedAxisTick />} />
+  <YAxis />
+  <Tooltip />
+</LineChart>
+);
+```
+
```
Comment (nikolasrieble) [inline storybook/stories/Getting started.mdx:126]: todo: Make this example show up.

**Comment by nikolasrieble** [comment]
@proke03 With this, we might not even need to migrate the website into the repository at all, since storybook could simply replace the website completely. 

What do you think?

**Comment by nikolasrieble** [comment]
While doing this, i realised that the code and the pictures on the https://recharts.org/en-US/guide/customize page never were in sync 👀 

Do we even want to keep them @ckifer ?

**Comment by ckifer** [comment]
lol 😅 - I think as long as we structure a few of out examples like this (i.e. can see the code, can see the result, can see a small blurb about what this example accomplishes). I think we are fine to remove this section

**Comment by nikolasrieble** [comment]
For now I will suggest to merge this, merely as an example of how to do docs, use code.
It also contains the Welcome page. 

Medium term this should go 😅

**Review by ckifer — APPROVED** [review]

**Comment by ckifer** [comment]
Yeah this all looks fine - probably a lot of things that don't need to be here but for the sake of copying existing docs

**Comment by proke03** [comment]
@nikolasrieble
If charts.org continues to play its role now, it can replace that completely.
However, I think the role of the site should be expanded. (forum, showcases)

File: storybook/stories/Welcome.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <CartesianGrid stroke="#eee" strokeDasharray="5 5" />
+  <XAxis dataKey="name" />
+  <YAxis />
+  <Line type="monotone" dataKey="uv" stroke="#8884d8" />
+  <Line type="monotone" dataKey="pv" stroke="#82ca9d" />
+  <Tooltip />
+</LineChart>
+
+```typescript jsx
+  <LineChart width={500} height={300} data={data}>
```
Comment (Yilun-Sun) [inline storybook/stories/Welcome.mdx:24]: I think we could indent one space to the left to make it has a better look 
image

**Review by Yilun-Sun — APPROVED** [review]

**Comment by nikolasrieble** [comment]
> @nikolasrieble If charts.org continues to play its role now, it can replace that completely. However, I think the role of the site should be expanded. (showcases, forum)

Could you please expand on what kind of cases you think could not live in storybook? A forum is a good example but I would doubt the value of it. After all we build a charting library, our users are engineers, and Github Discussions exist. 
If we focus on the core of Recharts, that will always be API documentation and actual feature work (and bug fixing 😅 )

**Comment by nikolasrieble** [comment]
> @nikolasrieble If charts.org continues to play its role now, it can replace that completely. However, I think the role of the site should be expanded. (showcases, forum)

Also, removing the website would massively reduce complexity, i.e. one thing less to deploy, less code to maintain, etc.

**Comment by nikolasrieble** [comment]
> Agreed, don't think we need a forum.
> 
> I don't think storybook results in documentation that I love 1000%, but it does result in readable and usable docs as long as we structure things correctly. 1 less thing to maintain (especially when that thing is old) is always beneficial to those maintaining.
> 
> My few concerns about totally removing the current website lie in 1. Branding - do our consumers care? And 2. Web domain - will we be able to get the storybook hosted as recharts.org?

Great question! 
re 1 - Branding wise I find storybook much more targeted to our community - developers. Although the welcome page should be improved design wise. 
re 2 - We most certainly can host it there.

**Comment by arcthur** [comment]
I agree that storybook is great document tool. but I want keep the brand, and use same style of website

**Pull request merged**